modules/dhcp_common/subnet.rb

```diff
@@ -10,16 +10,15 @@
 module Proxy::DHCP
   # Represents a DHCP Subnet
   class Subnet
-    attr_reader :network, :netmask, :server, :timestamp
+    attr_reader :network, :server, :timestamp
     attr_accessor :options
 
     include Proxy::DHCP
     include Proxy::Log
     include Proxy::Validations
 
-    def initialize network, netmask, options = {}
+    def initialize network, options = {}
```

Comment:
I think this class can be removed too in favour of ipv4 and ipv6-specific ones:
 * unused_ip code should be moved out from Subnet -- at least two providers have their own implementation. It's going to be ipv4/ipv6 specific too.
 * Everywhere else subnet class is either used as a container for options hash (which is broken) or for its ip address/mask